### DISCUSSION TITLE
Classroom Assignment will not finish configuration step

### DISCUSSION THREAD

**Discussion by Moticon** [community/community#43783] [GitHub Education] [answered]
I have created an assignment and shared the URL with my students.  The students were able to link their account and accept the assignment.

However, the assignment will never finish configuring. The message that the students are stuck on is "You accepted the assignment Lab1Starter. We're configuring your repository now. This may take a few minutes to complete. Refresh this page to see updates."\

The URL for the assignment is: https://classroom.github.com/a/R_1tnoL7
The classroom is: [Programming-II-Spring-2023-classroom-81868d](https://classroom.github.com/classrooms/121824958-programming-ii-spring-2023-classroom-81868d)

I used this all last year and had no problems, any help or pointers appreciated.

**EDIT** - Suddenly about 7:00 PM central time it started working again.  Was there a problem on the back end? 

Pat Smith

**Comment by sthompson-athenian** [comment]
I've been having this issue as well, intermittently yesterday and today. For projects in this repo: https://classroom.github.com/classrooms/121319546-athenian-apcs-s23

It's affected a number of students and myself, across multiple assignments. It seems like it drops in and out—there will be periods where no one can accept an assignment successfully.

**Reply by kurmasz** [reply]
I just now had a student report this same issue for https://classroom.github.com/classrooms/121507904-cis343-w23/assignments/c4c; 
however, about a dozen other students created repos without any problems.

**Reply by VictorNorman** [reply]
Having this problem right now with about 40 students.  (Hi, Zack!)

**Comment by Mr-Coxall** [comment]
I am also seeing the same problem.
I have checked 4 different assignments and all of them get stuck on the accept location.
I came back several minutes later and then they all started working.
I have never had this problem before.

**Reply by Mr-Coxall** [reply]
Correction, when I came back I had to "reaccept" them and then they just got stuck again.

**Comment by m2web** [comment]
I have students having this issue as well. Please advise.

**Reply by VictorNorman** [reply]
It is all working now again... But, too late for my labs yesterday. :-(

**Reply by kurmasz** [reply]
Do the missing repos just magically appear, or do the students need to manually "Accept again"?

**Comment by bgustin53** [comment]
Just adding another identical issue hoping someone from GitHub will chime it.  Also never had this problem before (7 year user) and this week had issue with 2 different repos on 2 different days, last Tuesday 1/10/2023 and today 1/13/2023

**Reply by fruittree** [reply]
We are having the same issue with some students can't accept the assignment. 3/27/2023.

**Comment by kurmasz** [comment]
My students who were having problems Wednesday and Thursday have all been able to create repos. However, I'm now getting reports from students in my other course who are just now trying to create repos and can't.

**Comment by m2web** [comment]
### Screen Shot from Student

For an FYI, here is a screenshot from a student on 1-13-23 at 3:49 PM EST attempting to create an assignment repo via an invitation link:

![reaccept](https://user-images.githubusercontent.com/289304/212477930-e70b3c5b-1945-4aef-890f-e2fb6ff870fa.jpg)

**Reply by m2web** [reply]
I have created a support ticket for this issue: https://support.github.com/ticket/personal/0/1966985

**Reply by kurmasz** [reply]
I opened this ticket Wednesday.  No (non-automated) reply yet.

https://support.github.com/ticket/personal/0/1960600

**Reply by jillhubbard** [reply]
Would love to upvote your ticket but can't locate it.  Is there a  direct link you could provide?  The /personal/0/# searches my own tickets and gives me a not found response since I didn't create it

**Comment by suhailrehman** [comment]
I just got hit by this issue; Students cannot accept assignments! I hope someone at Gtihub is alerted to this issue!

https://support.github.com/ticket/personal/0/1966496

**Comment by vnminin** [comment]
Same issue here. One student was able to create an assignment, but later no other student was able to replicate this success

**Comment by bgustin53** [comment]
Going on 3 days now without being able to create a repo from an assignment.  Has anybody come up with a work around other then hand making copies for all my students?  I am including the link to my support ticket: https://support.github.com/ticket/personal/0/1966360

**Reply by kurmasz** [reply]
In principle, it would be straightforward to write a program that uses the GitHub CLI and the GItHub API to create the desired repos in your Classroom's Organization.  The main limitation is that the assignments wouldn't appear in your classroom dashboard.  (But, you could still use a script to clone and pull all assignment repos.)

https://cli.github.com

**Reply by bgustin53** [reply]
Thank you.  I may do that if GClassroom remains unreliable.

**Comment by samus-aran** [comment]
Hi Everyone 👋 

Thank you for taking the time to share your experiences with the team. We've raised this issue with the internal team to investigate. We'll provide an update to this discussions when there is any further updates. Thank you all for your patience.

**Comment by Mr-Coxall** [comment]
I had this issue 5 days ago.
It then fixed itself.
Today it started up again.
No students can create a new Codespace again.

**Reply by chriswier** [reply]
Yep, ditto here as well.

**Comment by hdsbbrooks** [comment]
I am experiencing this problem as well.  Students that accept a new assignment just see it time out and eventually get the reaccept button, which then just loops the process.

**Comment by floan** [comment]
Adding myself to the thread to stay updated! @samus-aran please let us know when you have an estimate for how long it'll take to fix this issue so we can plan assignments accordingly. Thank you and good luck!

**Reply by floan** [reply]
My [support ticket](https://support.github.com/ticket/personal/0/1967555)!

**Comment by elizagriscom** [comment]
Most but apparently not all of my students are having this issue as well. I read on another thread about this same issue from July 2021 that said it was fixed (people suggested then that repositories needed to be public not private so I went and set up new assignments but this did not fix it for the students having issues. ) Please help. I also made a support ticket.

**Reply by elizagriscom** [reply]
I made a support ticket too support.github.com/ticket/1969585 @samus-aran  please let us know when you have an estimate for how long it'll take to fix this issue so we can plan assignments accordingly.

**Comment by niklas-hjelm** [comment]
I also added an issue on this topic [here](https://support.github.com/ticket/personal/0/1970413)

For me half of the Class had accepted the assignment before the issue appeared.

**Comment by hmm34** [comment]
I’m also experiencing this issue. Chiming in to get updates.

**Comment by caburu** [comment]
The same happens to me. Looking forward to a solution soon.

**Comment by dtg3** [comment]
I too am having this issue with my classes as well.

**Reply by dtg3** [reply]
I just tried forking my public templates from my template org to my classroom org, updated the assignment repo, and that fixed my problem with my test user. Anecdotal I know, but perhaps worth a shot.

**Reply by elizagriscom** [reply]
So glad you had success. I'm giving that a try- no luck at the moment with my test user. Did you keep your updated assignment repo public as well?  Did you keep the check on "Copy the main branch only
" before you clicked "Create Fork"? Thanks for sharing!

**Reply by elizagriscom** [reply]
Maybe they are working on it right now because weird things are happening! I made a new Classroom to try the fork idea and the assignments I am making keep disappearing. Time to walk away....

**Reply by dtg3** [reply]
Yes, I retained the original public template repo and just forked the assignment to the classroom org and changed the assignment to the forked template.

**Reply by elizagriscom** [reply]
Hi dtg3, sadly that did not work for me.  Please keep sharing ideas here if you have them!

**Comment by lassehav-oamk** [comment]
Same problem here. Too bad, this seemed like a great tool and started using it just last week. My students have reported two kinds of problems. First problem was the same what @m2web reported with the screenshot. Second problem for an new assignment created today is that the same as @Moticon reported "You accepted the assignment XXX. We're configuring your repository now. This may take a few minutes to complete. Refresh this page to see updates". The assignment repository is never created.

**Reply by elizagriscom** [reply]
We are having these same two problems. At a certain point in time, last week, a few of my students were able to accept the assignment and two even synced their code and submitted it (from VS Code).  Then I started having students seemingly connect to the Classroom but never get the assignments, with the "reaccept assignment" error message.  I teach two fully online CS courses that began this week and I'm at a loss as to how to proceed. Should I wait for a fix or make a last minute Plan B for this fully developed course that fully relies on GitHub Classroom?

**Reply by bgustin53** [reply]
This has been a week for me with 3 full CS classes.  I already made starter code and distributed it to one class.  If GitHub is going to support education, they should at least let us know... anything

**Reply by elizagriscom** [reply]
Agreed- please let us know soon GitHub Support. Us CS teachers are relying on you and need answers- or a PlanB - ASAP

**Comment by magitz** [comment]
Any update here from @github-staff would be great...
Unfortunately, having this happen at the start of the semester is particularly problematic. Students are already intimidated by new things and I'm trying to show them how easy it can be...only to fall flat on my face...
Any timeline? idea of what is going wrong? Had been working seamlessly for ages, and now it's broken...

**Reply by suhailrehman** [reply]
My best guess is that some update to GitHub Classroom to support codespaces broke existing functionality - particularly those who rely on public code templates from another org.

**Comment by TheTreeCode** [comment]
I'm a student. I just wanted to let everyone know that I've been having this same issue for the last few days and I just tried it again and it finally configured the repository for me! You all might want to ask your students to give it another shot now.

**Reply by floan** [reply]
Just worked for my students too! Not sure if its a proper fix or just flaky.

**Reply by elizagriscom** [reply]
Thank you for posting this TheTreeCode!  I just tried as well and it worked for me as test user. Fingers crossed for my students too!

**Comment by bgustin53** [comment]
It seems to be up again for me.  Fingers cross that there was a fix and not a fluke.

**Comment by samus-aran** [comment, marked as answer]
Hi everyone! 👋

We want to thank you all for sharing your experiences and helping us identify an issue that affected the ability to clone Classroom repos. The GitHub Classroom team is still investigating but we are happy to announce we have restored Classroom services to normal. We understand the impact this may have had at the beginning of the semester and we apologize for any inconveniences this has caused.

If you are still experiencing problems creating/accepting assignments and repos please do reply with what errors you are seeing and provide any information that would be relevant.

We appreciate your continued support of Classroom and GitHub Education.

**Reply by r-ahmed2022** [reply]
I am also facing the same problem, accepting the assignment, then waiting for the link, however, displaying error message

**Reply by cpena107** [reply]
I am also having a similar issue, several of my students are getting this error:
![PXL_20230714_162637144](https://github.com/community/community/assets/7307397/a36fd62d-2de2-4d82-ba4f-839a72902560)

**Reply by CWormley** [reply]
i have encountered this problem. I have checked my connection and restarted my computer. still the same response
<img width="680" alt="Screenshot 2023-08-23 at 11 40 32 AM" src="https://github.com/community/community/assets/119722180/7193fc4a-f14f-4d20-94c3-481a4391c412">

**Reply by zinstack625** [reply]
Assignment created, URL shared with students. Everyone can and did accept the assignment except one student. They have a public GitHub profile and is not any different from anyone else in terms of GitHub profile configuration as far as I can tell.

> There was an error accepting the assignment. We were not able to add the user to your assignment, please try again.. If this problem persists, please have the Classroom admin contact support.

And so, repository does not get created, and all the other stuff that usually gets set up.

So it persisted, and here I am. Am I even at the right place? Where should I poke and what should I look out for?

![image](https://github.com/community/community/assets/28690196/a17b89b9-e5f0-4d5a-bd85-a795cda5ba55)

**Comment by jillhubbard** [comment]
Thank you for fixing this issue!

**Comment by hlinero** [comment]
Thanks for the fix. Realy nice that everything is working now

**Comment by Moticon** [comment]
Please see Prof north. I am in Europe until MA 22.Sent from my Verizon, Samsung Galaxy smartphone
-------- Original message --------From: Riyaz Ahmed ***@***.***> Date: 5/11/23  3:05 PM  (GMT+00:00) To: community/community ***@***.***> Cc: Pat Smith ***@***.***>, Mention ***@***.***> Subject: Re: [community/community] Classroom Assignment will not finish configuration step (Discussion #43783) 
I am also facing the same problem, accepting the assignment, then waiting for the link, however, displaying error message

**Comment by vshimano** [comment]
My students encountered the same error message:
![20240206_094546](https://github.com/community/community/assets/156835512/d39966a9-c936-4f04-a5e4-f6b7b96c9f58)

**Reply by AminaEmenena** [reply]
My class is having the same issue as well 

`There was an error accepting the assignment. GitHub repository could not be created. Your instructor may need to allow forking of private repositories in the classroom's organization settings. If this problem persists, please have the Classroom admin contact support.`

**Reply by vshimano** [reply]
It would be really helpful if @github-staff was more responsive. Trying to introduce students to a new coding and collaborating environment is not an easy task by itself. When that environment does not work and there is no support, chances that students will use that platform in the future are very slim. And that's unfortunate.

**Reply by AminaEmenena** [reply]
Agreed. I am still having the issue and my students were really excited to use the tool.

**Comment by twennekers** [comment]
My students are experiencing an error with the same error message.
The template the code is drawn from is in a public repository.

**Comment by stevewds** [comment]
My students and I are experiencing the same error again today.

Accepting the assignment initially lands on an error "500" page. Then, trying to accept the assignment again results in a "We're configuring your repository now. This may take a few minutes to complete. Refresh this page to see updates." Which then fails after some 5 minutes or so.

It seems my students were able to accept the assignment about a hour ago (11am eastern time, New York) but not now (and I can't accept assignments now either).

**Reply by lukekenworthy-af** [reply]
Same here.

**Reply by mnwrhsn** [reply]
I’m also having the same issue for my class.

**Reply by stevevanhooser** [reply]
+1

**Reply by lukekenworthy-af** [reply]
@github-staff

**Reply by dowobeha** [reply]
> My students and I are experiencing the same error again today.
> 
> Accepting the assignment initially lands on an error "500" page. Then, trying to accept the assignment again results in a "We're configuring your repository now. This may take a few minutes to complete. Refresh this page to see updates." Which then fails after some 5 minutes or so.
> 
> It seems my students were able to accept the assignment about a hour ago (11am eastern time, New York) but not now (and I can't accept assignments now either).

I am also having this issue today (2024-09-03)

**Comment by reedcompbio** [comment]
I am having issues now too. Same as above: I get to the "you accepted the assignment" page, and after a few minutes it says there's an error.

![Screenshot 2024-09-03 at 10 08 34 AM](https://github.com/user-attachments/assets/d04a7a56-8fec-4a85-8656-7bf8c20e92fc)

I have tried this on firefox, chrome, and safari. The student is linked in the GitHub classroom. 

Is there a workaround for this?

**Reply by annaritz** [reply]
Adding my official GitHub username here (not the classroom tester).

**Reply by lukekenworthy-af** [reply]
Same here.

**Reply by ericaraujophd** [reply]
+1

**Reply by wangstagangstaa** [reply]
+1

**Reply by CraigGroup** [reply]
I have this issue for several students in my class of 82.  **I noticed that one difference is that the students affected don't have the automatic "add deadline" commit.**  The successfully accepted assignments have that commit automatically.  I expect this is just a sign the configuration failed.  We tried on chrome and Safari.   I am able to clone their repo, and I can see it on GitHub, but the student can not.

**Reply by chrissphinx** [reply]
@CraigGroup you can manually add them as a collaborator with write access, though that is pretty tedious and for 82 students 😬

**Reply by CraigGroup** [reply]
chris - do I add them to the organization?   Again, GitHub classroom created a repo, but it isn't properly configured.  I can see it, and even clone it, but the student cannot.

**Reply by chrissphinx** [reply]
@CraigGroup right, not to the organization, but for each students' repository you have to go in and add that particular student as an outside collaborator with write access to that specific repo. @rressler posted a screenshot that explains further: https://github.com/orgs/community/discussions/137628#discussioncomment-10536909

**Reply by annaritz** [reply]
Update: GitHub support seems to have rolled out a fix yesterday - I no longer have problems (though i had to delete previous repos and re-import the assignment in GitHub Classroom for it to work).

**Comment by chrissphinx** [comment]
digging up this thread too, will y'all set aside a dedicated form for this already? please? cc: @RyanHecht
